tools/capability-checker/src/types/model.ts

```diff
@@ -0,0 +1,30 @@
+import type {
+  LanguageModelV1,
+  ImageModelV1,
+  EmbeddingModelV1,
+} from '@ai-sdk/provider';
+import type { ModelCapabilities, Capability, ModelType } from './capability';
+
+export type ModelTestOptions =
+  | { type: 'language'; model: LanguageModelV1 }
+  | { type: 'image'; model: ImageModelV1 }
+  | { type: 'embedding'; model: EmbeddingModelV1<string> };
```

Comment:
I feel this can be simplified by including the test options here, but I might be mistaken:

```
  | { type: 'language'; model: LanguageModelV1, capabilities: LanguageModelCapabilities }
  | { type: 'image'; model: ImageModelV1, capabilities: ImageModelCapabilities }
  | { type: 'embedding'; model: EmbeddingModelV1<string>, ... };
```

In general I want to avoid generics whenever possible, and here it seems possible because there is a limited and known set of model types.